src/python/pants/backend/go/goals/tailor.py

```diff
@@ -62,22 +62,21 @@ async def find_putative_go_targets(
             PutativeTarget.for_target_type(
                 GoModTarget,
                 path=dirname,
-                name=os.path.basename(dirname),
+                name=None,
                 triggering_sources=sorted(filenames),
```

Comment:
Not for this change, but the fact everyone has to sort seems error-prone. Is there actually a case where not sorting makes sense?

Reply:
Agreed. I can't think of any reason not to sort.